zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/ZkClient.java

```diff
@@ -1849,13 +1924,23 @@ private void checkDataSizeLimit(byte[] data) {
   }
 
   public void watchForData(final String path) {
-    retryUntilConnected(new Callable<Object>() {
-      @Override
-      public Object call() throws Exception {
-        getConnection().exists(path, true);
-        return null;
+    watchForData(path, false);
+  }
+
+  private boolean watchForData(final String path, boolean skipWatchingNodeNotExist) {
+    try {
+      if (skipWatchingNodeNotExist) {
+        Stat stat = new Stat();
```

Comment:
This one is not used anyway, just new a Stat in the lamda.

Reply:
Newing a Stat is even not necessary. Null is totally fine.

Reply:
One more thing is, adding unused stat to this api also adds more metadata in the response packet. If we don't need it, I don't think it is a good idea to get the stat.

Reply:
As I said, relying on undocumented feature of API is not good practice. Also, can you verify they have the same behavior for all the old release?

Reply:
I think it is quite standard to pass a null to it if we don't need a stat. It is not guaranteed every use is in javadoc. The old release on GitHub: https://github.com/apache/zookeeper/blob/82186f0d00ebf30640404afcb55ef741f38678c2/src/java/test/org/apache/zookeeper/test/AuthTest.java#L45

Reply:
This is outdated. I added the comment as why we should not do it.